src/xcode/ENA/ENA/Source/Services/Risk/Provider/RiskProvider.swift

```diff
@@ -356,15 +357,21 @@ final class RiskProvider: RiskProviding {
 		appConfiguration: SAP_Internal_V2_ApplicationConfigurationIOS,
 		completion: @escaping (Result<[ExposureWindow], RiskProviderError>) -> Void
 	) {
+		guard exposureDetection == nil else {
+			// exposureDetection was already set and is therefor running so there is nothing to do here.
+			// in the future someone should debug why this funtion is called twice in the first place.
+			return
```

Comment:
Shouldn't we call the completion? Otherwise the call will be stuck.

Reply:
```suggestion
completion(.failure(.riskProviderIsRunning))
			return
```
Something like this?

Reply:
@marcussc I thought that too but I wasn't sure in this case. I will add it.